fuel-core/src/database/message.rs

```diff
@@ -12,37 +12,51 @@ use crate::{
 };
 use fuel_core_interfaces::{
     common::{
-        fuel_storage::Storage,
+        fuel_storage::{
+            StorageInspect,
+            StorageMutate,
+        },
         fuel_types::{
             Address,
             Bytes32,
             MessageId,
         },
     },
+    db::Messages,
     model::Message,
 };
 use std::{
     borrow::Cow,
     ops::Deref,
 };
 
-impl Storage<MessageId, Message> for Database {
+impl StorageInspect<Messages> for Database {
     type Error = KvStoreError;
 
+    fn get(&self, key: &MessageId) -> Result<Option<Cow<Message>>, KvStoreError> {
+        Database::get(self, key.as_ref(), Column::Messages).map_err(Into::into)
+    }
+
+    fn contains_key(&self, key: &MessageId) -> Result<bool, KvStoreError> {
+        Database::exists(self, key.as_ref(), Column::Messages).map_err(Into::into)
+    }
+}
+
+impl StorageMutate<Messages> for Database {
     fn insert(
         &mut self,
         key: &MessageId,
         value: &Message,
     ) -> Result<Option<Message>, KvStoreError> {
         // insert primary record
         let result =
-            Database::insert(self, key.as_ref(), columns::MESSAGES, value.clone())?;
+            Database::insert(self, key.as_ref(), Column::Messages, value.clone())?;
 
         // insert secondary record by owner
-        Database::insert(
+        let _: Option<bool> = Database::insert(
```

Comment:
Because I changed the `insert` method of the `Database`, it now requires explicit type annotation.

<img width="1287" alt="image" src="https://user-images.githubusercontent.com/18346821/190202417-3442f74c-eb46-4388-8b11-5d107750980b.png">

Reply:
hmm, would it make sense to have these database methods also use the new mappable types? That way each column is associated with a mappable, and we don't have to painstakingly set these return types in every call this way?

Reply:
We can, but we need to implement `Mappable` for semi-tables(like `(BlockHeight -> BlockHash)`).

Another problem is that `Database` works with other traits, like `Serialize` and `DeserializeOwned`. It is not a problem to add bounds, but all types used in the `Mappable::Key`, `Mappable::SetValue`, and `Mappable::GetValue` should implement them. It can't be true for all types and can cause problems in the future. But seems right now it is not a problem.

Reply:
Hmm yeah on second thought this might be overly constraining, as tables like Metadata don't have a uniform set of types between keys and values.